src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -388,6 +336,78 @@ await MovePoisonMessageToErrorQueue(receivedMessage, messageProcessingCancellati
             }
         }
 
+        public static TransportMessage ExtractTransportMessage(Message receivedMessage)
+        {
+            TransportMessage transportMessage;
+            string messageId;
+            var nativeMessageId = receivedMessage.MessageId;
+            if (receivedMessage.MessageAttributes.TryGetValue(Headers.MessageId, out var messageIdAttribute))
+            {
+                messageId = messageIdAttribute.StringValue;
+            }
+            else
+            {
+                messageId = nativeMessageId;
+            }
```

Comment:
TODO: lift this up from `ExtractTransportMessage` to the caller and pass in the `messageId` as an "override ID" in case no message ID can be found in any header.